src/pages/index.js

```diff
@@ -84,24 +90,148 @@ function handleImageClick(data) {
 function handleProfileEditSubmit(formValues) {
   userInfo.setUserInfo({
     name: formValues.title,
-    about: formValues.about,
+    about: formValues.description,
   });
+  editProfileModal.setLoading(true);
+  api
+    .setUserInfo(formValues.title, formValues.description)
+    .then((res) => {
+      userInfo.getUserInfo(res.name, res.about);
+      editProfileModal.setLoading(false);
+    })
+    .catch((err) => {
+      console.error("Error updating user info", err);
+      alert(err);
+    });
   editProfileModal.close();
 }
 function handleAddCardFormSubmit(formValues) {
   const name = formValues.title;
   const link = formValues.link;
+  // Make API request to upload card
+  addCardModal.setLoading(true);
+  api
+    .uploadCard({ name, link })
+    .then((cardData) => {
+      const card = createCard(cardData);
+      addCardModal.setLoading(false);
 
-  const card = createCard({ name, link });
-  cardList.addItem(card);
-  console.log(formValues);
-  cardAddForm.reset();
-  addCardModal.close();
+      cardList.addItem(card);
+      addCardModal.close();
+      cardAddForm.reset();
+    })
+    .catch((error) => {
+      console.error(error);
+    });
 }
 
 function createCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick);
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCardSubmit,
+    likeCard,
+    unlikeCard
+  );
   return card.generateCard();
 }
 
-//Event Listeners
+api
+  .getInitialCards()
+  .then((res) => {
+    console.log(res);
+    cardList.renderItems(res);
+  })
+
+  .catch((err) => alert(err));
+
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo(res);
+    userInfo.updateProfileImage(res);
+  })
+  .catch((err) => alert(err));
+
+//Avatar
+const profileImageForm = document.querySelector("#edit-avatar-form");
```

Comment:
**COULD BE IMPROVED**

We can get any form using their **unique** attribute `name` or `id` from `document.forms`

```js

const profileForm = document.forms["profile-form"];
const cardForm = document.forms["card-form"];

```

You need to add `name` or `id` first in `index.html` to achieve that.

More info here https://developer.mozilla.org/en-US/docs/Web/API/Document/forms